src/sage/rings/polynomial/multi_polynomial.pyx

```diff
@@ -871,16 +872,23 @@ cdef class MPolynomial(CommutativePolynomial):
 
         TESTS:
 
-        Check that :trac:`25022` is fixed::
+        Check that :issue:`25022` is fixed::
 
             sage: K.<x,y> = ZZ[]
             sage: (x*y).change_ring(SR).monomials()                                     # needs sage.rings.number_field sage.symbolic
             [x*y]
+
+        Check that :issue:`36832` is fixed::
+
+            sage: F = GF(11)
+            sage: phi = Hom(F,F).an_element()
+            sage: R.<x,y> = F[]
+            sage: x.change_ring(phi)
+            x
         """
         if isinstance(R, Map):
-        #if we're given a hom of the base ring extend to a poly hom
             if R.domain() == self.base_ring():
-                R = self.parent().hom(R, self.parent().change_ring(R.codomain()))
+                return self.map_coefficients(R, new_base_ring=R.codomain())
```

Comment:
```suggestion
                return self.map_coefficients(R)
```
The documentation of `map_coefficients` says:
```
If "f" is a :class:sage.categories.map.Map, then the polynomial will be defined over the codomain of "f".
```
Also, can `self.map_coefficients(R)` be used in any case (regardless of `R.domain() == self.base_ring()`) ?

See: #28541 for the corresponding bug and fix in polynomial rings (c615bb68304) and another way to reproduce with non-identity maps (fe1d9daf0ba).

The original code is from #20067 (1e6769852a). Note that the `change_ring()` method is repeated in `src/sage/rings/polynomial/multi_polynomial_element.py`. When is that one used?

Reply:
> The documentation of `map_coefficients` says:
> 
> ```
> If "f" is a :class:sage.categories.map.Map, then the polynomial will be defined over the codomain of "f".
> ```

Thanks. Done.

> Also, can `self.map_coefficients(R)` be used in any case (regardless of `R.domain() == self.base_ring()`) ?

Currently for the case `R.domain() != self.base_ring()`,  the code `R(self)` is used, and replacing this with  `self.map_coefficients(R)` would change the semantic. Changing the semantic is not in the scope of this PR.

> The original code is from #20067 ([1e67698](https://github.com/sagemath/sage/commit/1e6769852ac9b88c50832283014615ee8ed2b6ac)). Note that the `change_ring()` method is repeated in `src/sage/rings/polynomial/multi_polynomial_element.py`. When is that one used?

I don't understand your question. What is "that one"? Do you mean this one `return self.parent().change_ring(R)(self.dict())`? This one is used when `R` is a ring.

Reply:
Ah, you meant a different file `src/sage/rings/polynomial/multi_polynomial_element.py`. I will see.

Reply:
> > Also, can `self.map_coefficients(R)` be used in any case (regardless of `R.domain() == self.base_ring()`) ?
> 
> Currently for the case `R.domain() != self.base_ring()`, the code `R(self)` is used, and replacing this with `self.map_coefficients(R)` would change the semantic. Changing the semantic is not in the scope of this PR.

Looking it again, I am not sure if the original code `R(self)` for  the case `R.domain() != self.base_ring()` would really work for any meaningful cases. 

In the last commit, I removed it. Let's see how this would work out.

Reply:
Build & Test passed!